server/src/main/java/org/elasticsearch/action/support/PlainListenableActionFuture.java

```diff
@@ -1,99 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-package org.elasticsearch.action.support;
-
-import org.elasticsearch.action.ActionListener;
-import org.elasticsearch.action.ListenableActionFuture;
-
-import java.util.ArrayList;
-import java.util.List;
-
-public class PlainListenableActionFuture<T> extends AdapterActionFuture<T, T> implements ListenableActionFuture<T> {
-
-    volatile Object listeners;
-    boolean executedListeners = false;
-
-    protected PlainListenableActionFuture() {}
-
-    /**
-     * This method returns a listenable future. The listeners will be called on completion of the future.
-     * The listeners will be executed by the same thread that completes the future.
-     *
-     * @param <T> the result of the future
-     * @return a listenable future
-     */
-    public static <T> PlainListenableActionFuture<T> newListenableFuture() {
-        return new PlainListenableActionFuture<>();
-    }
-
-    @Override
-    public void addListener(final ActionListener<T> listener) {
-        internalAddListener(listener);
-    }
-
-    @Override
-    protected void done(boolean success) {
-        super.done(success);
-        synchronized (this) {
-            executedListeners = true;
-        }
-        Object listeners = this.listeners;
```

Comment:
This is the only volatile read outside of the mutex, but we just had the mutex anyway, may as well move this up a line and drop the `volatile`.